src/App.js

```diff
@@ -1,25 +1,87 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+import { useState } from "react";
+import Todo from "./components/Todo";
+import NewTodo from "./components/NewTodo";
+
+const initialTodos = [
+    {
+        id: 0,
+        text: "hello",
+        isDone: true,
+    },
+    {
+        id: 1,
+        text: "hellooo",
+        isDone: true,
+    },
+    {
+        id: 2,
+        text: "hello nah",
+        isDone: false,
+    },
+];
 
 function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+    const [newTodo, setNewTodo] = useState("");
+    const [allTodos, setAllTodos] = useState(initialTodos);
+
+    const handleTextChange = (e) => setNewTodo(e.target.value);
+
+    const createTodo = (e) => {
+        e.preventDefault();
+        const newAllTodos = [...allTodos];
+        newAllTodos.push({
+            id: allTodos.length,
+            text: newTodo,
+            isDone: false,
+        });
+        setAllTodos(newAllTodos);
+        setNewTodo("");
+    };
+
+    const handleIsDone = (id) => {
+        const newAllTodos = allTodos.map((item) => {
+            if (item.id === id) {
+                return { ...item, isDone: !item.isDone };
+            }
+            return item;
+        });
+        setAllTodos(newAllTodos);
+    };
+
+    const deleteTodo = (id) => {
+        const newAllTodos = [...allTodos];
```

Comment:
You do not need to deference array here because `.filter` creates new array
```suggestion
        const newAllTodos = [...allTodos];
```